src/app/components/containers/tasks.rs

```diff
@@ -67,15 +62,31 @@ impl Tasks {
 pub struct TasksProps<'a> {
     on: bool,
     tasks: &'a Vec<Task>,
+    filter: Box<dyn Fn(&&Task) -> bool>,
 }
 
 impl<'a> From<(&'a Model, &Context)> for TasksProps<'a> {
     fn from((model, context): (&'a Model, &Context)) -> TasksProps<'a> {
         let on = *context.stage() != Stage::SIDEBAR;
 
+        let filter = if *context.sidebar_stage() == SidebarStage::MENU {
```

Comment:
Let's change the stage function and sidebar_stage function to return a value instead of a reference. Performance shouldn't be much different. Code will be easier to follow.

Reply:
Done.